src/Engine/ProtoCore/Utils/ArrayUtils.cs

```diff
@@ -142,6 +142,44 @@ public static ClassNode GetGreatestCommonSubclassForArray(StackValue array, Runt
             return usageFreq;
         }
 
+        /// <summary>
+        /// Similar to GetTypeExamplesForLayer but it returns all arrays. Its purpose is to support
+        /// inspecting heterogeneous arrays in replication scenarios.
+        /// </summary>
+        /// <param name="array"></param>
+        /// <param name="runtimeCore"></param>
+        /// <returns></returns>
+        internal static List<StackValue> GetTypeExamplesForLayerWithoutArraySampling(StackValue array, RuntimeCore runtimeCore)
```

Comment:
So it looks like this method is not ignoring empty arrays while the other method with array sampling, is?

Reply:
Thinking about it, I could add it here too. It would not make a big difference for this one, as we are keeping all arrays, but getting the empty ones out of the way sooner may reduce the reduce params list size (no pun intended).